python/lsst/daf/butler/registry/queries/_results.py

```diff
@@ -304,7 +304,14 @@ def findDatasets(self, datasetType: Union[DatasetType, str], collections: Any, *
             components = [componentName]
         else:
             components = [None]
-        builder.joinDataset(datasetType, collections=collections, findFirst=findFirst)
+        if not builder.joinDataset(datasetType, collections=collections, findFirst=findFirst):
+            raise RuntimeError(
+                f"Error finding datasets of type {datasetType.name} in collections {collections}; "
+                "it is impossible for any such datasets to be found in any of those collections, "
+                "most likely because the dataset type is not registered.  "
+                "That this error may become a successful query that returns no results in the future, "
+                "because queries with no results are not usually considered an error."
```

Comment:
Should this be "Note that this error..."? As it is, the second sentence doesn't scan.